### ISSUE TITLE
Redesign main tab

### ISSUE THREAD

**Issue by tybug** [circleguard/circleguard#48]
We need to rewrite the Main tab to allow for more detects than just replay stealing. This new interface should be flexible enough to allow for more detects to be added, some with potentially strange/different requirements from current detects (think aim correct, which has more than a single number that needs to be set).

### Original Thoughts
```
[Main]    Results    Queue    Thresholds    Settings
[Check Map]    Screen User    Check Local    Verify    Visualize

Map id: [                                  ]

--------------- Steal Detect ---------------

Enabled [ ]

# everything below hidden since enabled unchecked
num: [     ]  span: [        ] mods: [     ]
user ids: [                                ]
threshold: [18]
--------------- Relax Detect ---------------

Enabled [x]
num: [             ]  span: [              ]
user ids: [                                ]
threshold: [50]

etc..
```

this is OK but doesn't give enough control without us making the interface near unusasble. For instance, how would you load two players with different mods? Or signify you want some replays in `replays1` and some in `replays2` for a steal check? Or compare across maps or users, or online to local? We historically handled these with special cases in the respective tabs but that feels increasingly unwieldy.

### Final Proposal
```
[Main]    Results    Queue    Thresholds    Settings

{Add Map} {Add User} {Add ReplayMap} {Add ReplayPath}

# Each {Add _} button would populate a draggable object onto the empty area below, 
# highlighted red to show not all necessary fields are filled out. It would look
# something like this, with each field being a LineEdit you could click and fill in:

|----------------|
| Map  id=1  {x} |
|----------------|
| map_id: [    ] |
| num:    [    ] |
| span:   [    ] |
| mods:   [    ] |
|----------------|


|---------------------|           |---------------|        etc.
| Steal Detect        |           | Relax Detect  |
|---------------------|           |---------------|
| Threshold: [    ]   |           | Threshold: [ ]|           
|---------------------|           |---------------|
| replays1 | replays2 |           |   draggable   |
|''''''''''|''''''''''|           |     area      |
| draggable| draggable|           |---------------|
|    area  |   area   |
|---------------------|


{---------------------  Run  --------------------}
```

The Map/User/etc can then be dragged onto the draggable areas, where they are referenced by id (Map 1) to avoid confusion between multiple of the same object, but do not dissapear from the main area, and can thus be dragged to multiple detects.

On run, of course, each detect with any objects in it is run by circlecore. This maps extremely well to circlecore, assuming circlecore == 2.5.0, as we represent each of core's objects as a user-creatable object in the gui.

no more subtabs below main, the user is given total freedom over how they want to compare maps/users/replays and what mods for each, and even if an individual object should be cached if we want to provide that as an option (might be useful).

If there was a need for it we could also provide functionality for adding more detects, so you can have a strict detect and a loose detect with different thresholds. But I think this is actually better served with a "suspicious threshold" and "cheat threshold" in the gui, and they're separated into two sections in the `Results` tab or soemthing. So you can still keep replays you may want to look at, but only want to get notified/really care about the blatant ones.

**Comment by InvisibleSymbol** [comment]
This is mock-up is the latest version of the design:
![](https://i.fiery.me/qLk2.png)

Some changes aren't in the image:
We will change the right side to work with 'blank' detects instead of single detects . Basically you add an empty detect and you can select which checks to run. Visualize would still be an extra item to add.

**Comment by InvisibleSymbol** [comment]
done c:

**Comment by tybug** [comment]
(closed with #57)
